### PR TITLE
[REVIEW] Port libcudf strings wrap api to cython/python

### PR DESCRIPTION
**Pull request by galipremsagar**
This implements one api(`wrap`) in : #3992 
Had to do it as a separate PR due to git issues to pull changes in the original PR: #4311

### CODE REVIEW

File: python/cudf/cudf/core/column/string.py [hunk 1]
```diff
@@ -689,6 +690,63 @@ def split(self, pat=None, n=-1, expand=True, **kwargs):
...
+            raise NotImplementedError(
+                "`break_long_words` parameter is not supported"
+            )
+
+        if "break_on_hyphens" in kwargs:
+            raise NotImplementedError(
+                "`break_on_hyphens` parameter is not supported"
+            )
+
+        if "expand_tabsbool" in kwargs:
```
Comment (beckernick) [inline python/cudf/cudf/core/column/string.py]: This is covered above

File: python/cudf/cudf/core/column/string.py [hunk 2]
```diff
@@ -689,6 +690,63 @@ def split(self, pat=None, n=-1, expand=True, **kwargs):
...
+            raise NotImplementedError(
+                "`expand_tabsbool` parameter is not supported"
+            )
```
Comment (kkraus14) [inline python/cudf/cudf/core/column/string.py]:
```suggestion
```

**Comment by kkraus14** [comment]
@galipremsagar looks like we're getting failures in the tests for `wrap`

**Comment by galipremsagar** [comment]
> @galipremsagar looks like we're getting failures in the tests for `wrap`

Logged an issue: https://github.com/rapidsai/cudf/issues/4348

and commented out the edge case for now.

Added some notes to the function doc which you might want to check?

File: python/cudf/cudf/core/column/string.py [hunk 3]
```diff
@@ -710,6 +710,21 @@ def wrap(self, width, **kwargs):
...
+
+            expand_tabs = False
+
+            replace_whitespace = True
+
+            drop_whitespace = True
+
+            break_long_words = False
+
+            break_on_hyphens = False
```
Comment (kkraus14) [inline python/cudf/cudf/core/column/string.py]: Are these the Pandas defaults? If so we should throw a warning loud and clear if someone tries to diverge from here. If these aren't the default we should throw a warning that we don't match default behavior.

Comment (galipremsagar) [reply python/cudf/cudf/core/column/string.py]: These are pandas optional params, our current behavior resembles that particular optional parameters setting. They are not defaults, still would you want me throw a warning ? or raise error if they pass the param and it diverges from that specific value?

Comment (kkraus14) [reply python/cudf/cudf/core/column/string.py]: According to this: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.str.wrap.html the defaults are:
```
expand_tabs: True
replace_whitespace: True
drop_whitespace: True
break_long_words: True
break_on_hyphens: True
```

So given we're not matching the defaults we should throw a warning that indicate where our current defaults differ from Pandas defaults, as well as throwing exceptions if someone tries to explicitly set something that we don't support yet.

Comment (galipremsagar) [reply python/cudf/cudf/core/column/string.py]: @kkraus14, done.

File: python/cudf/cudf/core/column/string.py [hunk 4]
```diff
@@ -730,30 +730,42 @@ def wrap(self, width, **kwargs):
...
-        if "expand_tabsbool" in kwargs:
-            raise NotImplementedError(
-                "`expand_tabsbool` parameter is not supported"
-            )
+        expand_tabs = kwargs.get("expand_tabs", False)
+        if expand_tabs:
+            raise NotImplementedError("`expand_tabs=True` is not supported")
+        warnings.warn(
+            "wrap current implementation defaults to `expand_tabs`=False"
+        )
```
Comment (kkraus14) [inline python/cudf/cudf/core/column/string.py]: Should follow a similar pattern for other kwargs where we should warn if our default value differs from Pandas and then we should throw if the user explicitly passes a value we don't yet support.
```suggestion
        expand_tabs = kwargs.get("expand_tabs", None)
        if expand_tabs == True:
            raise NotImplementedError("`expand_tabs=True` is not supported")
        elif expand_tabs is None:
            warnings.warn(
                "wrap current implementation defaults to `expand_tabs`=False"
            )
```

Comment (galipremsagar) [reply python/cudf/cudf/core/column/string.py]: Made similar changes for the rest of the parameter settings.

File: python/cudf/cudf/core/column/string.py [hunk 5]
```diff
@@ -730,42 +730,57 @@ def wrap(self, width, **kwargs):
...
+        replace_whitespace = kwargs.get("replace_whitespace", None)
         if not replace_whitespace:
             raise NotImplementedError(
                 "`replace_whitespace=False` is not supported"
             )
+        elif replace_whitespace is None:
+            warnings.warn(
+                "wrap current implementation defaults to \
+                    `replace_whitespace`=True"
+            )
```
Comment (kkraus14) [inline python/cudf/cudf/core/column/string.py]: We already have the correct default here:
```suggestion
        replace_whitespace = kwargs.get("replace_whitespace", True)
        if not replace_whitespace:
            raise NotImplementedError(
                "`replace_whitespace=False` is not supported"
            )
```

File: python/cudf/cudf/core/column/string.py [hunk 6]
```diff
@@ -730,42 +730,57 @@ def wrap(self, width, **kwargs):
...
+        drop_whitespace = kwargs.get("drop_whitespace", None)
         if not drop_whitespace:
             raise NotImplementedError(
                 "`drop_whitespace=False` is not supported"
             )
+        elif drop_whitespace is None:
+            warnings.warn(
+                "wrap current implementation defaults to \
+                    `drop_whitespace`=True"
+            )
```
Comment (kkraus14) [inline python/cudf/cudf/core/column/string.py]: Here too:
```suggestion
        drop_whitespace = kwargs.get("drop_whitespace", True)
        if not drop_whitespace:
            raise NotImplementedError(
                "`drop_whitespace=False` is not supported"
            )
```

**Comment by galipremsagar** [comment]
> Two minor changes

Resolved. 👍

**Pull request merged**